acceptance/tests/resource/zone/ticket_4840_should_create_zone_with_zpool.rb

```diff
@@ -4,32 +4,46 @@
 require 'puppet/acceptance/solaris_util'
 extend Puppet::Acceptance::ZoneUtils
 
+def poolsetup(agent)
+  on agent,"mkdir /tstzones"
+  on agent,"mkfile 2500m /tstzones/dsk"
```

Comment:
Yes it is necessary. I tried small files, but that much space is need. It is rather slow and extends the test time considerably, but it does what is required in the ticket.